frontend/src/FullPageRoutes/Profile.tsx

```diff
@@ -21,14 +20,23 @@ const useStyles = makeStyles({
     },
 });
 
+type UserProfile = {
+    first_name: string;
+    last_name: string;
+    phone_number?: string;
+    age?: number;
+    gender: string;
+
+}
+
 export default function Profile() {
     const classes = useStyles();
 
     const [username, setUsername] = React.useState<string>(null);
     React.useEffect(() => {
-        request<User>('/api/current_user/', 'get', true, false).then(
-            (user) => {
-                setUsername(user.parsedBody.username);
+        request<UserProfile>('/api/profile/', 'get', true).then(
+            (profile) => {
+                setUsername(profile.parsedBody.first_name);
```

Comment:
I'm considering changing UserProfile to this after I pull these changes. what do you think?

```
type UserProfile = {
    username: string
    data: {
        first_name: string
        last_name: string
        phone_number: string
        age: string | number
        gender: string
    }
}
```

Reply:
and then removing the setUsername hook.

Reply:
I feel a more elegant solution would be to use a better serialization strategy on the backend

Reply:
agreed